app/scripts/modules/core/src/search/infrastructure/infrastructureSearch.service.ts

```diff
@@ -1,65 +1,46 @@
 import { module, IDeferred, IPromise, IQService } from 'angular';
 import { Observable, Subject } from 'rxjs';
 
-import { UrlBuilderService, URL_BUILDER_SERVICE } from 'core/navigation';
+import { URL_BUILDER_SERVICE } from 'core/navigation';
 import { ProviderServiceDelegate, PROVIDER_SERVICE_DELEGATE } from 'core/cloudProvider';
 
-import { getFallbackResults, ISearchResult, ISearchResults, SearchService, SEARCH_SERVICE } from '../search.service';
-import { IResultDisplayFormatter, ISearchResultType, searchResultTypeRegistry, } from '../searchResult/searchResultsType.registry';
-import { externalSearchRegistry } from '../externalSearch.registry';
+import { InfrastructureSearchServiceV2 } from './infrastructureSearchV2.service';
+import { ISearchResult, SEARCH_SERVICE } from '../search.service';
+import { SearchResultType, searchResultTypeRegistry, } from '../searchResult';
 import { SearchStatus } from '../searchResult/SearchResults';
 
-export interface ISearchResultSet<T extends ISearchResult = any> {
-  type: ISearchResultType;
+export interface ISearchResultSet<T extends ISearchResult = ISearchResult> {
+  type: SearchResultType;
   results: T[];
   status: SearchStatus;
   error?: any;
 }
 
 export interface IProviderResultFormatter {
-  [category: string]: IResultDisplayFormatter,
+  [category: string]: (entry: ISearchResult, fromRoute?: boolean) => string;
 }
 
 export class InfrastructureSearcher {
   private deferred: IDeferred<ISearchResultSet[]>;
   public querySubject: Subject<string> = new Subject<string>();
 
-  constructor(private $q: IQService, private providerServiceDelegate: ProviderServiceDelegate, searchService: SearchService, urlBuilderService: UrlBuilderService) {
-    this.querySubject.switchMap(
-      (query: string) => {
+  constructor(
+    private $q: IQService,
+    private providerServiceDelegate: ProviderServiceDelegate,
+    infrastructureSearchServiceV2: InfrastructureSearchServiceV2,
+  ) {
+    this.querySubject
+      .switchMap((query: string) => {
         if (!query || query.trim() === '') {
-          return Observable.of(getFallbackResults());
+          const fallbackResults = searchResultTypeRegistry.getAll()
+            .map(type => ({ type, results: [], status: SearchStatus.INITIAL } as ISearchResultSet));
+          return Observable.of(fallbackResults);
         }
-        return Observable.zip(
-          searchService.search({ q: query, type: searchResultTypeRegistry.getSearchCategories() }),
-          externalSearchRegistry.search({ q: query }).toArray(),
-          (s1: ISearchResults<any>, s2: ISearchResultSet[]) => {
-            s1.results = s2.reduce((acc, srs: ISearchResultSet) => acc.concat(srs.results), s1.results);
-            return s1;
-          }
-        )
+        return infrastructureSearchServiceV2.search({ key: query }).toArray();
+      })
+      .subscribe((result: ISearchResultSet[]) => {
+        this.deferred.resolve(result);
```

Comment:
I don't understand what this code is doing well enough, but it feels weird that the searcher is going to resolve with some value here - does it not always return the same value from the promise once it's resolved it once?

Reply:
I was picking my battles and didn't refactor this part.  The searcher creates new `this.deferred` each time the query changes.  It returns the new promise to the caller.  When results are ready, it resolves the "latest" promise with the data.